### ISSUE TITLE
Installation on macOS has issues

### ISSUE THREAD

**Issue by swvanderlaan** [davidgohel/gdtools#49]
Hi,

This is related to https://github.com/davidgohel/gdtools/issues/24, https://github.com/davidgohel/gdtools/issues/40, and https://github.com/davidgohel/gdtools/issues/43. I reviewed all three issues.

Here's my information:

- I did have macOS Mojave, `R`, `brew` etc all up and running, but I needed a clean install anyway of macOS Mojave.
- I installed `R` and other packages via `brew`. Until just last week this never caused any issues, but admittedly something somewhere has changed.
- I have installed CommandLine Tools, and executed this `` command because apparently this is needed nowadays to have headers properly set.
- I installed - before even installing `R` - `brew install pkg-config cairo`.

Nevertheless I get this error when I do `install.packages("gdtools"):

```
* installing *source* package ‘gdtools’ ...
** package ‘gdtools’ successfully unpacked and MD5 sums checked
Found pkg-config cflags and libs!
Using PKG_CFLAGS=
Using PKG_LIBS=-L/usr/local/Cellar/cairo/1.16.0/lib -lcairo
------------------------- ANTICONF ERROR ---------------------------
Configuration failed because cairo was not found. Try installing:
 * deb: libcairo2-dev (Debian, Ubuntu)
 * rpm: cairo-devel (Fedora, CentOS, RHEL)
 * csw: libcairo_dev (Solaris)
 * brew: cairo (OSX)
If cairo is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a cairo.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
--------------------------------------------------------------------
ERROR: configuration failed for package ‘gdtools’
* removing ‘/usr/local/Cellar/r/3.5.2_2/lib/R/library/gdtools’
```

So `cairo` is not found! Weird, because it is really there, as is XQuartz:

```
cairo: stable 1.16.0 (bottled), HEAD
Vector graphics library with cross-device output support
https://cairographics.org/
/usr/local/Cellar/cairo/1.16.0 (117 files, 5.5MB) *
  Poured from bottle on 2019-03-14 at 07:05:11
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/cairo.rb
==> Dependencies
Build: pkg-config ✔
Required: fontconfig ✔, freetype ✔, glib ✔, libpng ✔, pixman ✔
==> Options
--HEAD
	Install HEAD version

brew cask info xquartz
xquartz: 2.7.11 (auto_updates)
https://www.xquartz.org/
/usr/local/Caskroom/xquartz/2.7.11 (74.6MB)
From: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/xquartz.rb
==> Name
XQuartz
==> Artifacts
XQuartz.pkg (Pkg)
```
My system is:

```
brew config
HOMEBREW_VERSION: 2.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0f07fe5c5cdfdf0f424bba7b879faa38aedc8ee4
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6de36908d29d90d0d0f88708a6cd8362e3ca0190
Core tap last commit: 14 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.dedbyK1ghF/org.macosforge.xquartz:0
HOMEBREW_LOGS: /Users/swvanderlaan/Library/Logs/Homebrew
CPU: quad-core 64-bit skylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.21.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.2
macOS: 10.14.3-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
CLT headers: 10.1.0.0.1.1539992718
XQuartz: 2.7.11 => /opt/X11
```

_However_, when I run as per your suggestion in https://github.com/davidgohel/gdtools/issues/24 this code: `devtools::install_github("davidgohel/gdtools")`. Now it does work!

```
devtools::install_github("davidgohel/gdtools")
Downloading GitHub repo davidgohel/gdtools@master
   checking for file ‘/private/var/folders/hk/7x9pbh313hz8hyfdbwqkgc4c0000gn/T/RtmpMh3rzm/remotes40e277ff9ae/davidgohel-gdtools-6707d8✔  checking for file ‘/private/var/folders/hk/7x9pbh313hz8hyfdbwqkgc4c0000gn/T/RtmpMh3rzm/remotes40e277ff9ae/davidgohel-gdtools-6707d80/DESCRIPTION’ (343ms)
─  preparing ‘gdtools’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  running ‘cleanup’
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘gdtools_0.1.7.9000.tar.gz’

* installing *source* package ‘gdtools’ ...
Found pkg-config cflags and libs!
Using PKG_CFLAGS=-I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16
Using PKG_LIBS=-L/usr/local/Cellar/cairo/1.16.0/lib -lcairo
** libs
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c CairoContext.cpp -o CairoContext.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c RcppExports.cpp -o RcppExports.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c font_metrics.cpp -o font_metrics.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c gdtools.cpp -o gdtools.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c raster_to_base64.cpp -o raster_to_base64.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c sys_fonts.cpp -o sys_fonts.o
clang++  -I"/usr/local/Cellar/r/3.5.2_2/lib/R/include" -DNDEBUG -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/cairo/1.16.0/include/cairo -I/usr/local/Cellar/glib/2.60.0_1/include -I/usr/local/Cellar/glib/2.60.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.60.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/pixman/0.38.0/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.13.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/libpng/1.6.36/include/libpng16 -I../inst/include/ -I"/usr/local/Cellar/r/3.5.2_2/lib/R/library/Rcpp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include   -fPIC  -g -O2  -c versions.cpp -o versions.o
clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -L/usr/local/Cellar/r/3.5.2_2/lib/R/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -o gdtools.so CairoContext.o RcppExports.o font_metrics.o gdtools.o raster_to_base64.o sys_fonts.o versions.o -L/usr/local/Cellar/cairo/1.16.0/lib -lcairo -L/usr/local/Cellar/r/3.5.2_2/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
installing to /usr/local/Cellar/r/3.5.2_2/lib/R/library/gdtools/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (gdtools)
```

`cairo` is found. 

I hope this will help you in debugging.

Best,

Sander

**Comment by wcass77** [comment]
I was having the same problem and using `devtools::install_github("davidgohel/gdtools")` worked for me. Thanks!

**Comment by jeroen** [comment]
This is fixed in gdtools 1.8.0 via https://github.com/davidgohel/gdtools/commit/7b0a89fc07e77b35d67492ae088afb06b2919022
